libs/header/test/testing.go

```diff
@@ -0,0 +1,287 @@
+package test
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"encoding/binary"
+	"encoding/gob"
+	"errors"
+	"fmt"
+	"math"
+	"testing"
+	"time"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"golang.org/x/crypto/sha3"
+
+	"github.com/celestiaorg/celestia-node/libs/header"
+)
+
+type Raw struct {
+	ChainID      string
+	PreviousHash header.Hash
+
+	Height int64
+	Time   time.Time
+}
+
+type DummyHeader struct {
+	Raw
+
+	hash header.Hash
+}
```

Comment:
Q: why are two separate structs and not a single combination?

Reply:
To expose `ChainID()`, `Time()`, `Height()`, `PreviousHash()` methods, while also being able to easily assign values for those fields outside of `test` package.